homeassistant/components/rfxtrx/__init__.py

```diff
@@ -115,14 +121,24 @@ def handle_receive(event):
         for subscriber in RECEIVED_EVT_SUBSCRIBERS:
             subscriber(event)
 
-    device = config[DOMAIN][ATTR_DEVICE]
+    device = config[DOMAIN].get(ATTR_DEVICE)
+    host = config[DOMAIN].get(CONF_HOST)
+    port = config[DOMAIN].get(CONF_PORT)
     debug = config[DOMAIN][ATTR_DEBUG]
     dummy_connection = config[DOMAIN][ATTR_DUMMY]
 
     if dummy_connection:
         rfx_object = rfxtrxmod.Connect(
             device, None, debug=debug, transport_protocol=rfxtrxmod.DummyTransport2
         )
+    elif port is not None:
+        # If port is se then we create a TCP connection
```

Comment:
If port is set then we create a TCP connection.

Reply:
Fixed